cinema/views.py

```diff
@@ -56,3 +78,47 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    @staticmethod
+    def _params_to_int(qs):
+        return [int(str_id) for str_id in qs.split(",")]
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = (
+                queryset.select_related("cinema_hall", "movie").annotate(
+                    tickets_available=F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                )
+            )
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+        if movie is not None:
+            queryset = queryset.filter(movie__id=movie)
+        if date is not None:
+            queryset = queryset.filter(
+                show_time__year=date[0:4],
+                show_time__month=date[5:7],
+                show_time__day=date[8:10],
```

Comment:
It's too bad to use indexes to split a date into parts
use strptime function
And filter by date directly